shared/crypto/keypair.go

```diff
@@ -135,6 +136,26 @@ func (kp *encKeyPair) Unmarshal(bz []byte) error {
 	return nil
 }
 
+// UnmarshalJSON Unmarshals a JSON string into an encKeyPair struct
+func (ekp *encKeyPair) UnmarshalJSON(data []byte) error {
+	type Alias encKeyPair
+	aux := &struct {
```

Comment:
This temporary struct is very similar to what was there before but I think its outside the scope of this PR to add a Protobuf for the KeyPair.

Reply:
right, yeah I think we're at the limits of what go can do for us with json marshalling and custom types.